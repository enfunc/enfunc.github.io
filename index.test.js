const server = require('http-server');

const serve = (port, root = './') => {
  const s = server.createServer({root: root})
  s.listen(port)
  return s
}

const content = {
  name: 'Enfunc',
  desc: "Enfunc provides top-notch software engineering, advisory, and funding support to help unlock your business's full potential.",
  url: "https://www.enfunc.com",
  logo: "https://www.enfunc.com/media/enfunc.png",
  title: 'Enfunc | Top-tier software, creativity, and zest.',
  services: [' We code', ' We advise', ' We invest'],
  formUrl: 'https://formspree.io/f/moqzkrzl',
  logos: 12,
  cta: " Let's chat",
  accentColor: "rgb(178, 239, 193)",
  complementColor: "rgb(239, 178, 224)",
}

const getText = async (selector) => await page.$eval(selector, el => el.textContent);
const getAttr = async (selector, attr) => await page.$eval(selector, (el, attr) => el.getAttribute(attr), attr);
const getStyleProperty = async (selector, prop) => await page.$eval(selector, (el, prop) => getComputedStyle(el).getPropertyValue(prop), prop);

describe('Enfunc', () => {
  let localhost;
  let baseUrl;

  beforeAll(async () => {
    const port = 8081;
    localhost = serve(port);
    baseUrl = `http://localhost:${port}`;
    await page.goto(baseUrl);
  });

  afterAll(() => {
    if (localhost) localhost.close();
  })

  test('has the correct metadata', async () => {
    const tags = {
      'property="og:title"': content.title,
      'name="description"': content.desc,
      'property="og:description"': content.desc,
      'property="og:image"': content.logo,
      'property="og:url"': content.url,
      'property="og:type"': 'website',
    };
    for (const tag of Object.keys(tags)) {
      const content = await getAttr(`meta[${tag}]`, 'content');
      expect(content).toBe(tags[tag]);
    }
  });

  test('has the correct title', async () => {
    const title = await getText('title');
    expect(title).toBe(content.title);
  });

  test('has the correct webmanifest', async () => {
    const link = await getAttr('link[rel="manifest"]', 'href');
    expect(link).toBe('site.webmanifest');
  });

  test('has the correct favicons', async () => {
    const tags = {
      'link[rel="icon"]': 'media/favicon.ico',
      'link[rel="apple-touch-icon"]': 'media/apple-touch-icon.png',
    }
    for (const tag of Object.keys(tags)) {
      const link = await getAttr(tag, 'href');
      expect(link).toBe(tags[tag]);
    }
  });

  test('has the correct ld+json', async () => {
    const keys = {
      name: content.name,
      description: content.desc,
      url: content.url,
      logo: content.logo,
    }
    const ldjson = await page.evaluate(() => {
      const tags = document.querySelectorAll('script[type="application/ld+json"]');
      if (tags.length > 1) throw new Error('unexpected count');
      return JSON.parse(tags[0].textContent);
    });
    for (const key of Object.keys(keys)) {
      expect(ldjson[key]).toBe(keys[key]);
    }
  });

  test('has the expected CSS', async () => {
    const css = [
      'https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css',
      'https://fonts.googleapis.com/css2?family=Bayon&family=Schibsted+Grotesk&display=swap',
      `${baseUrl}/style.min.css`
    ];
    const hrefs = await page.$$eval('link[rel="stylesheet"]', els => els.map((el) => el.href));
    for (const href of hrefs) {
      expect(css).toContain(href);
    }
  });

  test('has the expected JS', async () => {
    const js = [
      'https://enfunc.containers.piwik.pro/ppms.js',
      'https://browser.sentry-cdn.com/7.50.0/bundle.tracing.replay.min.js',
      'https://enfunc.containers.piwik.pro/4466da6e-a361-42fc-8ea2-e93fc3e9c75b.js',
      'https://kit.fontawesome.com/21547d334e.js',
      'https://js.sentry-cdn.com/6d72b914e64e4d3c8488bbc1d22a2f75.min.js',
      `${baseUrl}/script.min.js`,
    ];
    const srcs = await page.$$eval('script', els => els.map((el) => el.src));
    for (const src of srcs) {
      if (src) expect(js).toContain(src);
    }
  });

  test('has the correct header', async () => {
    const h1 = await getText('h1');
    expect(h1).toBe(content.name);

    const bg = await getStyleProperty('header', 'background-color');
    expect(bg).toBe(content.accentColor);
  });

  test('has the correct services', async () => {
    const headings = await page.$$eval('.services h2', els => els.map(el => el.textContent));
    expect(headings).toEqual(content.services);

    const paras = await page.$$eval('.services p', els => els.map(el => el.textContent));
    paras.forEach((p) => {
      expect(p.length).toBeGreaterThan(280);
    })
  });

  const ctaSelector = '.cta-button';

  test('has the correct CTA button', async () => {
    const bg = async () => await getStyleProperty(ctaSelector, 'background-color');

    const text = await getText(ctaSelector);
    expect(text).toBe(content.cta);

    expect(await bg()).toBe(content.accentColor);
    await page.hover(ctaSelector);
    expect(await bg()).toBe(content.complementColor);
  });

  test('has the expected CTA functionality', async () => {
    const display = async () => await page.$eval('.contact-overlay', el => el.style.display);

    expect(await display()).toBe('');
    await page.click(ctaSelector);
    expect(await display()).toBe('block');
    await page.click('.close');
    expect(await display()).toBe('none');
  });

  test('has the correct contact form', async () => {
    const action = await getAttr('form', 'action');
    expect(action).toBe(content.formUrl);

    const method = await getAttr('form', 'method');
    expect(method).toBe('POST');
  });

  test('has the correct number of logos', async () => {
    const logos = await page.$$('.logos img');
    expect(logos.length).toBe(content.logos);
  });

  test('has the correct footer', async () => {
    const text = await getText('footer p');
    expect(text.length).toBeGreaterThan(400);

    const year = new Date().getFullYear();
    const copy = await getText('.copy');
    expect(copy).toBe(`Â© ${year} ${content.name}. All rights reserved.`);
  });
});
