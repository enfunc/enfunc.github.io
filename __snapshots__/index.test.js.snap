// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enfunc matches snapshot 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="Enfunc is a global tech consultancy specializing in tailored digital solutions, advisory, startup guidance, and funding, enabling businesses to thrive in the digital era.">

  <meta property="og:title" content="Enfunc | Unleashing potential, accelerating progress">
  <meta property="og:description" content="Enfunc is a global tech consultancy specializing in tailored digital solutions, advisory, startup guidance, and funding, enabling businesses to thrive in the digital era.">
  <meta property="og:image" content="https://www.enfunc.com/media/enfunc.png">
  <meta property="og:url" content="https://www.enfunc.com">
  <meta property="og:type" content="website">

  <title>Enfunc | Unleashing potential, accelerating progress</title>

  <link rel="manifest" href="data:application/manifest+json;base64,eyJuYW1lIjoiRW5mdW5jIiwic2hvcnRfbmFtZSI6IkVuZnVuYyIsImRlc2NyaXB0aW9uIjoiRW5mdW5jIGlzIGEgZ2xvYmFsIHRlY2ggY29uc3VsdGFuY3kgc3BlY2lhbGl6aW5nIGluIHRhaWxvcmVkIGRpZ2l0YWwgc29sdXRpb25zLCBhZHZpc29yeSwgc3RhcnR1cCBndWlkYW5jZSwgYW5kIGZ1bmRpbmcsIGVuYWJsaW5nIGJ1c2luZXNzZXMgdG8gdGhyaXZlIGluIHRoZSBkaWdpdGFsIGVyYS4iLCJ0aGVtZV9jb2xvciI6IiNCMkVGQzEiLCJiYWNrZ3JvdW5kX2NvbG9yIjoiI0ZGRkZGRiIsImRpc3BsYXkiOiJzdGFuZGFsb25lIiwib3JpZW50YXRpb24iOiJwb3J0cmFpdCIsInNjb3BlIjoiLyIsInN0YXJ0X3VybCI6Ii8iLCJpY29ucyI6W3sic3JjIjoibWVkaWEvYW5kcm9pZC1jaHJvbWUtMTkyeDE5Mi5wbmciLCJzaXplcyI6IjE5MngxOTIiLCJ0eXBlIjoiaW1hZ2UvcG5nIn0seyJzcmMiOiJtZWRpYS9hbmRyb2lkLWNocm9tZS01MTJ4NTEyLnBuZyIsInNpemVzIjoiNTEyeDUxMiIsInR5cGUiOiJpbWFnZS9wbmcifV19">
  <link rel="icon" href="media/favicon.ico" sizes="any">
  <link rel="apple-touch-icon" href="media/apple-touch-icon.png">

  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Bayon&family=Oswald&family=Playfair+Display&display=swap">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/modern-normalize@2.0.0/modern-normalize.min.css">
  <style>*{box-sizing:border-box}body{background:url('/media/bg.webp');background-size:cover;color:#000;width:100vw;height:100vh;overflow-x:hidden}main{display:flex;align-items:center;height:100vh;width:100vw}section{background:#b2efc1;margin:3em;padding:3em;width:75%;opacity:.9;box-shadow:0 1px 2px 0 rgb(60 64 67 / 30%),0 4px 8px 4px rgb(60 64 67 / 15%)}a,h1,h2{margin:0;padding:0;text-decoration:none}a{font-family:Oswald,sans-serif;color:#000}h1{font-family:Bayon,cursive;font-size:10em}h2{font-family:Oswald,sans-serif;font-size:4em;margin-top:-.5em}p{font-family:'Playfair Display',serif;font-size:1.67em}.logos{margin-top:3em;display:flex;justify-content:space-evenly;gap:1em;flex-flow:row wrap}img{max-height:2.5em}a:hover,img:hover{opacity:.75}@media screen and (max-width:800px){h1{font-size:8em}h2{font-size:3em}section{width:95%;margin:1em;padding:1em}}</style>

  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "Organization",
  "name": "Enfunc",
  "url": "https://www.enfunc.com",
  "logo": "https://www.enfunc.com/media/enfunc.png",
  "description": "Enfunc is a global tech consultancy specializing in tailored digital solutions, advisory, startup guidance, and funding, enabling businesses to thrive in the digital era.",
  "contactPoint": {
    "@type": "ContactPoint",
    "email": "hello@enfunc.com"
  }
}
  </script>
</head>
<body>
<main>
  <section>
    <h1>Enfunc</h1>
    <h2>Unleashing potential, accelerating progress</h2>
    <p>
      Our technology empowers millions across industries and borders daily.
      Say hi at <a href="mailto:hello@enfunc.com">hello@enfunc.com</a>.
    </p>
    <div class="logos">
      <a href="https://www.starsgroup.com/" target="_blank"><img src="media/logo_stars.webp" alt="The Stars Group" title="The Stars Group"></a><a href="https://www.lloydsbankinggroup.com/" target="_blank"><img src="media/logo_lloyds.webp" alt="Lloyds Banking Group" title="Lloyds Banking Group"></a><a href="https://www.nhs.uk/" target="_blank"><img src="media/logo_nhs.webp" alt="NHS" title="NHS"></a><a href="https://www.pfizer.com/" target="_blank"><img src="media/logo_pfizer.webp" alt="Pfizer" title="Pfizer"></a><a href="https://auth0.com/" target="_blank"><img src="media/logo_auth0.webp" alt="Auth0" title="Auth0"></a><a href="https://www.array.com/" target="_blank"><img src="media/logo_array.webp" alt="Array" title="Array"></a><a href="https://www.celtra.com/" target="_blank"><img src="media/logo_celtra.webp" alt="Celtra" title="Celtra"></a>
    </div>
  </section>
</main>
<script>
if (window.location.origin.includes("enfunc.com")) {
  (function(window, document, dataLayerName, id) {
    window[dataLayerName]=window[dataLayerName]||[],window[dataLayerName].push({start:(new Date).getTime(),event:"stg.start"});var scripts=document.getElementsByTagName('script')[0],tags=document.createElement('script');
    function stgCreateCookie(a,b,c){var d="";if(c){var e=new Date;e.setTime(e.getTime()+24*c*60*60*1e3),d="; expires="+e.toUTCString();f="; SameSite=Strict"}document.cookie=a+"="+b+d+f+"; path=/"}
    var isStgDebug=(window.location.href.match("stg_debug")||document.cookie.match("stg_debug"))&&!window.location.href.match("stg_disable_debug");stgCreateCookie("stg_debug",isStgDebug?1:"",isStgDebug?14:-1);
    var qP=[];dataLayerName!=="dataLayer"&&qP.push("data_layer_name="+dataLayerName),isStgDebug&&qP.push("stg_debug");var qPString=qP.length>0?("?"+qP.join("&")):"";
    tags.async=!0,tags.src="https://enfunc.containers.piwik.pro/"+id+".js"+qPString,scripts.parentNode.insertBefore(tags,scripts);
    !function(a,n,i){a[n]=a[n]||{};for(var c=0;c<i.length;c++)!function(i){a[n][i]=a[n][i]||{},a[n][i].api=a[n][i].api||function(){var a=[].slice.call(arguments,0);"string"==typeof a[0]&&window[dataLayerName].push({event:n+"."+i+":"+a[0],parameters:[].slice.call(arguments,1)})}}(i[c])}(window,"ppms",["tm","cm"]);
  })(window, document, 'dataLayer', '4466da6e-a361-42fc-8ea2-e93fc3e9c75b');
}
</script>
</body>
</html>
"
`;
